### ISSUE TITLE
Chrome 100: BrowserNav V2.3 generates huge delays and hiccups on certain websites

### ISSUE THREAD

**Issue by amirsol81** [mltony/nvda-browser-nav#8]
With BrowserNav V2.3, moving around on some websites or simply typing characters into certain edit boxes makes NVDA silent or quite sluggish. For instance, do the following:
1. Open https://www.merriam-webster.com
2. Move to the first edit box: Search.
3. We want to search for the word "language," so type l, a, and n.
4. At this point, NVDA, regardless of its version, should be quite sluggish or unresponsive for a few seconds. Continuing typing will make it more sluggish even if it regains speech.
This doesn't happen with BrowserNav disabled.
As another, perhaps more obvious example, do the following:
1. Go to the following page: https://www.baranbax.com/
2. It's a Persian website, but it doesn't matter. As you land on the page and start navigating by H among headings, NVDA becomes quite sluggish, and even Chrome 100 displays the "Does not respond" message. Again, disabling BrowserNave takes care of the issue.
Even using the Search facility in Chrome 100's History window, Control+H, generates such a delay as Chrome finds results based on what we type into the box.

**Comment by akash07k** [comment]
Yes, I confirm this issue.
Happens with me too.
unfortunately I had to disable BrowserNav for now.
I hope it gets fixed soon

**Comment by Surveyor123** [comment]
Hi,
Confirmed here as well. But the issue is not limited to chrome 100; it can be reproduced with both Firefox version 98.0.2 and Microsoft Edge version 100.0.1185.29 either.

**Comment by amirsol81** [comment]
@Surveyor123 Confirmed with MS Edge 100, but I can also duplicate these hiccups with Mozilla Thunderbird  V91.7.0. That is, when I read emails in the Message window and press Up/Down to move past links, NVDA stops responding for a few seconds as it reaches some links. Disabling BrowserNav V2.3 resolves the issue there, too.

**Comment by Surveyor123** [comment]
Until Tony hits the stage downgrading to 2.2 may be an option.

**Comment by amirsol81** [comment]
@Surveyor123 Any chance of providing a V2.2 link to me? I'd like to downgrade to V2.2 for now if it takes care of the problem.

**Comment by amirsol81** [comment]
@Surveyor123 I now have V2.2 but it also has some issues. For instance, close Chrome 100 and re-open it, and go to https://translate.google.com
With BrowserNav V2.2, NVDA becomes silent and it takes almost 10 second in order for the screen reader to resume speech and present the loaded page. Of course, V2.2 is less problematic than V2.3 so it's seemingly worth keeping enabled for now.

**Comment by Surveyor123** [comment]
No release yet, but all issues mentioned here seem resolved with recent code update. Thanks Tony!

**Comment by amirsol81** [comment]
@mltony The latest commit has apparently fixed all Chrome 100/non-Chrome issues. However, now NVDA says, "document," as each page gets opened. Without BrowserNav we don't hear this "document" announcement for newly loaded pages. Not a major issue but I thought I'd report it here. And, once more, thanks for your awesome efforts related to BrowserNav! It's a true gamechanger.

**Comment by mltony** [comment]
I believe I fixed the issue in v2.4. I'll close this for now, but feel free to reopen if any problems are still not resolved.
